packages/nextjs/src/config/loaders/valueInjectionLoader.ts

```diff
@@ -0,0 +1,26 @@
+import { LoaderThis } from './types';
+
+type LoaderOptions = {
+  values: Record<string, unknown>;
+};
+
+/**
+ * Set values on the global/window object at the start of a module.
+ *
+ * Options:
+ *   - `values`: A record where the keys correspond to the keys of the global values to set and the values
+ *        correspond to the values of the values on the global object. Values must be JSON serializable.
+ */
+export default function valueInjectionLoader(this: LoaderThis<LoaderOptions>, userCode: string): string {
+  // We know one or the other will be defined, depending on the version of webpack being used
+  const { values } = 'getOptions' in this ? this.getOptions() : this.query;
+
+  // Define some global proxy that works on server and on the browser.
+  let injectedCode = 'var _sentryCollisionFreeGlobalObject = typeof window === "undefined" ? global : window;\n';
+
+  Object.entries(values).forEach(([key, value]) => {
+    injectedCode += `_sentryCollisionFreeGlobalObject.${key} = ${JSON.stringify(value)};\n`;
```

Comment:
M: It might be safer to use backets here, just in case `key` has a space or a dash in it. (I mean, I know, we're the ones in charge of what the keys are going to be, so we could just avoid that, but this would prevent someone in the future from not realizing that restriction and running into a problem.)

```suggestion
    injectedCode += `_sentryCollisionFreeGlobalObject["${key}"] = ${JSON.stringify(value)};\n`;
```